### PR TITLE
Example of random_mask_pixel to perform biased random crop

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds an example (notebook) of a usage of ```ops.RandomMaskPixel``` operator to generate crop centers that are placed on foreground pixels of a segmentation mask with a given probability.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Extended "COCO Reader with augmentations" notebook to show how to perform a biased or balanced random crop, by specifying the probability of the crop window to be centered at a foreground pixel in the segmentation mask.*
     *Fixed seed generation in RandomMaskPixel to be done per-sample instead of per-thread, for deterministic results*
 - Affected modules and functionalities:
     *RandomMaskPixel operator*
 - Key points relevant for the review:
     *All*
 - Validation and testing:
     *Notebook*
 - Documentation (including examples):
     *Notebook*

**JIRA TASK**: *[DALI-1723]*

### CODE REVIEW

File: dali/operators/segmentation/random_mask_pixel.cc [hunk 1]
```diff
@@ -78,11 +78,11 @@ class RandomMaskPixelCPU : public Operator<CPUBackend> {
 
 RandomMaskPixelCPU::RandomMaskPixelCPU(const OpSpec &spec)
     : Operator<CPUBackend>(spec),
-      seed_(spec.GetArgument<int64_t>("seed")),
+      rngs_(spec.GetArgument<int64_t>("seed"), spec.GetArgument<int64_t>("batch_size")),
```
Comment (JanuszL) [inline dali/operators/segmentation/random_mask_pixel.cc:81]: It won't play nicely with our recent movement through variable batch size.
How about the old approach, but:
```
auto &rng = rng_[sample_idx%thread_id];
```
So you still have only `thread_pool.size()` rngs, and have deterministic way of assigning them to the threads.

**Review by mzient — APPROVED** [review]

File: dali/operators/segmentation/random_mask_pixel.cc (hunk 1 above)
Comment (jantonguirao) [reply dali/operators/segmentation/random_mask_pixel.cc:81]: It will place nicely since I am creating enough seeds to fit the maximum batch size.

Comment (JanuszL) [reply dali/operators/segmentation/random_mask_pixel.cc:81]: So be it.

**Review by JanuszL — APPROVED** [review]

**Comment by jantonguirao** [comment]
!builds

**Comment by jantonguirao** [comment]
!build

**Pull request merged**